virttest/qemu_vm.py

```diff
@@ -448,10 +448,25 @@ def add_serial(devices, name, filename):
             if '86' in params.get('vm_arch_name', arch.ARCH):
                 cmd += " -device isa-serial"
             elif 'ppc' in params.get('vm_arch_name', arch.ARCH):
-                cmd += " -device spapr-vty"
-                # Workaround for console issue, details:
-                #   lists.gnu.org/archive/html/qemu-ppc/2013-10/msg00129.html
-                cmd += _add_option("reg", "0x30000000")
+                if ('tcp'in params.get('pro_backend')):
+                    pro_host = params.get('pro_host')
+                    pro_port = params.get('pro_port')
+                    cmd  = ""
+                    cmd += " -chardev socket"
+                    cmd += _add_option("id", serial_id)
+                    cmd += _add_option("host", pro_host)
+                    cmd += _add_option("port", pro_port)
+                    cmd += _add_option("server", "NO_EQUAL_STRING")
+                    cmd += _add_option("nowait", "NO_EQUAL_STRING")
+                    cmd += " -device spapr-vty"
+                    # Workaround for console issue, details:
+                    # lists.gnu.org/archive/html/qemu-ppc/2013-10/msg00129.html
+                    cmd += _add_option("reg", "0x30000000")
+                else:
+                    cmd += " -device spapr-vty"
+                    # Workaround for console issue, details:
+                    # lists.gnu.org/archive/html/qemu-ppc/2013-10/msg00129.html
+                    cmd += _add_option("reg", "0x30000000")
```

Comment:
Thanks for you so quickly feedback @CongLi 
And I will double check it,thanks a lot.